src/rprblender/nodes/blender_nodes.py

```diff
@@ -972,42 +973,48 @@ def export_hybrid(self):
         return shader2
 
 
-class ShaderNodeTexCoord(RuleNodeParser):
+class ShaderNodeTexCoord(NodeParser):
     # outputs: Generated, Normal, UV, Object, Camera, Window, Reflection
     # Supported outputs by RPR: Normal, UV
 
-    nodes = {
-        "Generated": {
-            "type": pyrpr.MATERIAL_NODE_INPUT_LOOKUP,
-            "params": {
-                pyrpr.MATERIAL_INPUT_VALUE: pyrpr.MATERIAL_NODE_LOOKUP_UV,
-            },
-            "warn": "TexCoord Generated output is not supported, UV will be used"
-        },
-        "Normal": {
-            "type": pyrpr.MATERIAL_NODE_INPUT_LOOKUP,
-            "params": {
+    def export(self):
+        tex_coord_type = self.socket_out.name
+
+        if tex_coord_type == 'Generated':
+            data = self.create_node(pyrpr.MATERIAL_NODE_INPUT_LOOKUP, {
+                pyrpr.MATERIAL_INPUT_VALUE: pyrpr.MATERIAL_NODE_LOOKUP_P_LOCAL,
+            })
+            if self.object:
+                # normalize over object bounding box
+                # get min and max of boinding box
+                min = tuple(min(p[i] for p in self.object.bound_box) for i in range(3))
```

Comment:
Have following error here:
```
  File "D:\amd-gpuopen\RadeonProRenderBlenderAddon\src\rprblender\nodes\blender_nodes.py", line 990, in export
    min = tuple(min(p[i] for p in self.object.bound_box) for i in range(3))
  File "D:\amd-gpuopen\RadeonProRenderBlenderAddon\src\rprblender\nodes\blender_nodes.py", line 990, in <genexpr>
    min = tuple(min(p[i] for p in self.object.bound_box) for i in range(3))
NameError: free variable 'min' referenced before assignment in enclosing scope
```
Better not to override built-in min and max functions, suggest to use names mn, mx.

Reply:
fixed